google-cloud-firestore/lib/google/cloud/firestore/watch/listener.rb

```diff
@@ -0,0 +1,297 @@
+
+            # Customize the provided initial listen request
+            init_listen_req = @init_listen_req.dup.tap do |req|
+              req.add_target.resume_token = String(@inventory.resume_token)
+              req.add_target.read_time = \
+                Convert.time_to_timestamp(@inventory.read_time)
+              req.add_target.target_id = 0x42
+            end
+
+            # Always create a new enum queue
+            synchronize do
+              @request_queue = EnumeratorQueue.new self
+              @request_queue.push init_listen_req
+            end
+
+            # Not an @var, we get a new enum each time
+            enum = synchronize do
+              @client.service.listen @request_queue.each
+            end
+
+            loop do
+
+              # Break loop, close thread if stopped
+              break if synchronize { @stopped }
+
+              begin
+                # Cannot syncronize the enumerator, causes deadlock
+                response = enum.next
+
+                case response.response_type
+                when :target_change
+                  case response.target_change.target_change_type
+                  when :NO_CHANGE
+                    # No change has occurred. Used only to send an updated
+                    # +resume_token+.
+
+                    @inventory.persist(
+                      response.target_change.resume_token,
+                      Convert.timestamp_to_time(
+                        response.target_change.read_time
+                      )
+                    )
+
+                    if @inventory.current? && @inventory.changes?
+                      synchronize do
+                        send_callback @inventory.build_query_snapshot
+                      end
+                    end
+                  when :CURRENT
+                    # The targets reflect all changes committed before the targets
+                    # were added to the stream.
+                    #
+                    # This will be sent after or with a +read_time+ that is
+                    # greater than or equal to the time at which the targets were
+                    # added.
+                    #
+                    # Listeners can wait for this change if read-after-write
+                    # semantics are desired.
+
+                    @inventory.persist(
+                      response.target_change.resume_token,
+                      Convert.timestamp_to_time(
+                        response.target_change.read_time
+                      )
+                    )
+
+                    @inventory.current = true
+                  when :RESET
+                    # The targets have been reset, and a new initial state for the
+                    # targets will be returned in subsequent changes.
+                    #
+                    # After the initial state is complete, +CURRENT+ will be
+                    # returned even if the target was previously indicated to be
```

Comment:
s/returned/reset/?

Reply:
This is the documentation copied from the protobuf. It is copied here for reference only.

Reply:
IOW I don’t know if reset or returned is correct.